// firebase-tools/lib/mcp/tools/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.availableTools = void 0;
const index_js_1 = require("./auth/index.js");
const index_js_2 = require("./dataconnect/index.js");
const index_js_3 = require("./firestore/index.js");
const index_js_4 = require("./directory/index.js");
const index_js_5 = require("./core/index.js");
const index_js_6 = require("./storage/index.js");
function availableTools(fixedRoot, activeFeatures) {
    const toolDefs = addPrefixToToolName("firebase_", index_js_5.coreTools);
    if (!fixedRoot) {
        toolDefs.push(...index_js_4.directoryTools);
    }
    if (!activeFeatures || !activeFeatures.length) {
        activeFeatures = Object.keys(tools);
    }
    for (const key of activeFeatures) {
        toolDefs.push(...tools[key]);
    }
    return toolDefs;
}
exports.availableTools = availableTools;
const tools = {
    firestore: addPrefixToToolName("firestore_", index_js_3.firestoreTools),
    auth: addPrefixToToolName("auth_", index_js_1.authTools),
    dataconnect: addPrefixToToolName("dataconnect_", index_js_2.dataconnectTools),
    storage: addPrefixToToolName("storage_", index_js_6.storageTools),
};
function addPrefixToToolName(prefix, tools) {
    return tools.map((tool) => (Object.assign(Object.assign({}, tool), { mcp: Object.assign(Object.assign({}, tool.mcp), { name: `${prefix}${tool.mcp.name}` }) })));
}
