// firebase-tools/lib/mcp/tools/directory/get_project_directory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.get_project_directory = void 0;
const zod_1 = require("zod");
const tool_js_1 = require("../../tool.js");
const util_js_1 = require("../../util.js");
const detectProjectRoot_js_1 = require("../../../detectProjectRoot.js");
exports.get_project_directory = (0, tool_js_1.tool)({
    name: "get_project_directory",
    description: "Gets the current Firebase project directory. If this has been set using the `set_project_directory` tool it will return that, otherwise it will look for a PROJECT_ROOT environment variable or the current working directory of the running Firebase MCP server.",
    inputSchema: zod_1.z.object({}),
    annotations: {
        title: "Get Firebase Project Directory",
        readOnlyHint: true,
    },
}, (_, { host }) => {
    if (!(0, detectProjectRoot_js_1.detectProjectRoot)({ cwd: host.projectRoot }))
        return Promise.resolve((0, util_js_1.mcpError)(`There is no detected 'firebase.json' in directory '${host.projectRoot}'. Please use the 'set_project_directory' tool to activate a Firebase project directory.`));
    return Promise.resolve((0, util_js_1.toContent)(`The current Firebase project directory is '${host.projectRoot}'.`));
});
