// firebase-tools/lib/mcp/tools/auth/get_user.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.get_user = void 0;
const zod_1 = require("zod");
const tool_js_1 = require("../../tool.js");
const util_js_1 = require("../../util.js");
const auth_js_1 = require("../../../gcp/auth.js");
exports.get_user = (0, tool_js_1.tool)({
    name: "get_user",
    description: "Retrieves a user based on an email address, phone number, or UID.",
    inputSchema: zod_1.z.object({
        email: zod_1.z.string().optional(),
        phoneNumber: zod_1.z.string().optional(),
        uid: zod_1.z.string().optional(),
    }),
    annotations: {
        title: "Get information about 1 user.",
        readOnlyHint: true,
    },
    _meta: {
        requiresAuth: true,
        requiresProject: true,
    },
}, async ({ email, phoneNumber, uid }, { projectId }) => {
    if (email === undefined && phoneNumber === undefined && uid === undefined) {
        return (0, util_js_1.mcpError)(`No user identifier supplied in auth_get_user tool`);
    }
    return (0, util_js_1.toContent)(await (0, auth_js_1.findUser)(projectId, email, phoneNumber, uid));
});
