// firebase-tools/lib/mcp/tools/storage/get_download_url.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.get_object_download_url = void 0;
const zod_1 = require("zod");
const tool_js_1 = require("../../tool.js");
const util_js_1 = require("../../util.js");
const storage_js_1 = require("../../../gcp/storage.js");
exports.get_object_download_url = (0, tool_js_1.tool)({
    name: "get_object_download_url",
    description: "Retrieves the download URL for an object in Firebase Storage.",
    inputSchema: zod_1.z.object({
        bucket: zod_1.z.string().nullish().describe("The bucket name in Firebase Storage."),
        objectPath: zod_1.z
            .string()
            .describe("The path to the object in Firebase storage without the bucket name attached"),
    }),
    annotations: {
        title: "Get the download url for an obejct in Firebase Storage.",
        readOnlyHint: true,
    },
    _meta: {
        requiresProject: true,
        requiresAuth: true,
    },
}, async ({ bucket, objectPath }, { projectId }) => {
    if (!bucket) {
        bucket = `${projectId}.firebasestorage.app`;
    }
    const downloadUrl = await (0, storage_js_1.getDownloadUrl)(bucket, objectPath);
    return (0, util_js_1.toContent)(downloadUrl);
});
