// firebase-tools/lib/mcp/tools/dataconnect/get_connector.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.get_connector = void 0;
const zod_1 = require("zod");
const tool_js_1 = require("../../tool.js");
const util_js_1 = require("../../util.js");
const client = require("../../../dataconnect/client.js");
const fileUtils_js_1 = require("../../../dataconnect/fileUtils.js");
const converter_js_1 = require("./converter.js");
exports.get_connector = (0, tool_js_1.tool)({
    name: "get_connector",
    description: "Get the Firebase Data Connect Connectors in the project, which includes the pre-defined GraphQL queries accessible to client SDKs.",
    inputSchema: zod_1.z.object({
        serviceId: zod_1.z
            .string()
            .nullable()
            .describe("The Firebase Data Connect service ID to look for. If there is only one service defined in firebase.json, this can be omitted and that will be used."),
    }),
    annotations: {
        title: "Obtain the Firebase Data Connect Connectors that's available in the backend",
        readOnlyHint: true,
    },
    _meta: {
        requiresProject: true,
        requiresAuth: true,
    },
}, async ({ serviceId }, { projectId, config }) => {
    const serviceInfo = await (0, fileUtils_js_1.pickService)(projectId, config, serviceId || undefined);
    const connectors = await client.listConnectors(serviceInfo.serviceName, ["*"]);
    return (0, util_js_1.toContent)(connectors.map(converter_js_1.connectorToText).join("\n\n"));
});
